### PR TITLE
cshatag: init at 2019-12-03

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Silent data corruption paranoia

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change (none)
- [x] Tested execution of all binary files
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/cshatag/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = with lib; {
+    description = "A tool to detect silent data corruption";
+    homepage = "https://github.com/rfjakob/cshatag";
+    license = licenses.mit;
+    platform = platforms.linux;
+  };
+
+}
+
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cshatag/default.nix:34]:
```suggestion
}
```

GitHub does not show the final new line does it? If it does I did not mean to remove it.

File: pkgs/os-specific/linux/cshatag/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoPackage, fetchFromGitHub, attr }:
+
+buildGoPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cshatag/default.nix:3]: Would you mind switching to buildGoModule? It does not requires the deps.nix.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102102` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>cshatag</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102102` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/os-specific/linux/cshatag/default.nix (hunk 2 above)
Comment (rnhmjoj) [reply pkgs/os-specific/linux/cshatag/default.nix:3]: `buildGoModule` builds a Go module, but this isn't one. This package has only two dependencies, so deps.nix cluttering Nixpkgs isn't much of a concern, here.
Also see https://github.com/NixOS/nixpkgs/issues/84826 for the downsides of `buildGoModule`.

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/cshatag/default.nix:3]: I didn't check if it is a Go module or not. Thanks for the explanation.

File: pkgs/os-specific/linux/cshatag/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  postInstall = ''
+    cd go/src/${goPackagePath}
+    make install $makeFlags
+  '';
+
+  meta = with lib; {
+    description = "A tool to detect silent data corruption";
+    homepage = "https://github.com/rfjakob/cshatag";
+    license = licenses.mit;
+    platform = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cshatag/default.nix:30]:
```suggestion
    platforms = platforms.linux;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/cshatag/default.nix (hunk 3 above)
Comment (rnhmjoj) [reply pkgs/os-specific/linux/cshatag/default.nix:30]: Thank you, I always seem to make this typo.

File: pkgs/os-specific/linux/cshatag/default.nix (hunk 1 above)
Comment (rnhmjoj) [reply pkgs/os-specific/linux/cshatag/default.nix]: Removed, thank you.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102102` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cshatag</li>
  </ul>
</details>

**Comment by rnhmjoj** [comment]
I removed `attr` from the builds inputs, which I forgot to do earlier. This is needed to run the test suite but it's not working: the tests need user extended attributes which are not availalble in the Nix build sandbox.

**Pull request merged**